pkg/cloud/data/result.go

```diff
@@ -0,0 +1,253 @@
+	}
+	return nil
+}
+
+func (r *CloudResultRepository) EndExecution(ctx context.Context, execution testkube.Execution) error {
+	req := EndExecutionResultRequest{Execution: execution}
+	_, err := execute(ctx, r.cloudClient, CmdResultEndExecution, req, r.apiKey)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *CloudResultRepository) GetLabels(ctx context.Context) (map[string][]string, error) {
+	response, err := execute(ctx, r.cloudClient, CmdResultGetLabels, nil, r.apiKey)
+	if err != nil {
+		return nil, err
+	}
+	var commandResponse GetLabelsResultResponse
+	if err := json.Unmarshal(response.Response, &commandResponse); err != nil {
+		return nil, err
+	}
+	return nil, nil
+}
+
+func (r *CloudResultRepository) DeleteByTest(ctx context.Context, testName string) error {
+	req := DeleteByTestResultRequest{TestName: testName}
+	_, err := execute(ctx, r.cloudClient, CmdResultDeleteByTest, req, r.apiKey)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *CloudResultRepository) DeleteByTestSuite(ctx context.Context, testSuiteName string) error {
+	req := DeleteByTestSuiteResultRequest{TestSuiteName: testSuiteName}
+	_, err := execute(ctx, r.cloudClient, CmdResultDeleteByTestSuite, req, r.apiKey)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *CloudResultRepository) DeleteAll(ctx context.Context) error {
+	req := DeleteAllResultRequest{}
+	_, err := execute(ctx, r.cloudClient, CmdResultDeleteAll, req, r.apiKey)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *CloudResultRepository) DeleteByTests(ctx context.Context, testNames []string) error {
+	req := DeleteByTestsResultRequest{TestNames: testNames}
+	_, err := execute(ctx, r.cloudClient, CmdResultDeleteByTests, req, r.apiKey)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *CloudResultRepository) DeleteByTestSuites(ctx context.Context, testSuiteNames []string) error {
+	req := DeleteByTestSuitesResultRequest{TestSuiteNames: testSuiteNames}
+	_, err := execute(ctx, r.cloudClient, CmdResultDeleteByTestSuites, req, r.apiKey)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *CloudResultRepository) DeleteForAllTestSuites(ctx context.Context) error {
+	req := DeleteForAllTestSuitesResultResponse{}
+	_, err := execute(ctx, r.cloudClient, CmdResultDeleteForAllTestSuites, req, r.apiKey)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *CloudResultRepository) GetTestMetrics(ctx context.Context, name string, limit, last int) (testkube.ExecutionsMetrics, error) {
+	req := GetTestMetricsResultRequest{Name: name, Limit: limit, Last: last}
+	response, err := execute(ctx, r.cloudClient, CmdResultGetTestMetrics, req, r.apiKey)
+	if err != nil {
+		return testkube.ExecutionsMetrics{}, err
+	}
+	var commandResponse GetTestMetricsResultResponse
+	if err := json.Unmarshal(response.Response, &commandResponse); err != nil {
+		return testkube.ExecutionsMetrics{}, err
+	}
+	return commandResponse.Metrics, nil
+}
```

Comment:
Looks like Generics will be useful to deduplicate this code. At least in the future)